gateway/gateway_test.go

```diff
@@ -252,15 +251,21 @@ func TestStartCLogging(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
+	// Ensure logging is started if there is no entry in the loggingRoutineStarted map.
 	g.startCLogging(ctx)
 	test.That(t, loggingRoutineStarted["test-gateway"], test.ShouldBeTrue)
 	test.That(t, g.workers, test.ShouldNotBeNil)
 
-	// Test case 2: Ensure no new goroutine is started if the loggingRoutineStarted entry is true.
-	// reset g.workers to test for new workers
+	// Test that closing the gateway removes the gateway from the loggingRoutineStarted map.
+	err := g.Close(ctx)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, len(loggingRoutineStarted), test.ShouldEqual, 0)
```

Comment:
We should be equivalent in our tests - test that worker are nil in this case, and  also that that logginroutine length is 1 in the test above.

Reply:
`Close` doesn't set the workers as nil it just stops them, I dont think theres a way to test if they are running or stopped so I was using `ShouldBeNil` as kind of a workaround to test for that, but it wouldn't work here.

Reply:
I could also just add a `g.workers = nil` line to the Close function but it wouldn't be necessary for anything except the test.

Reply:
Hmm, that is surprising since I assumed by using the `sync.WaitGroup`'s done function, `StoppableWorker`'s `cancel` function would decrement the workers count in its implementation in goutils. But if it does not then feel free to just check the length for parity between the two test cases.

I'll have to think a little about the `stoppableworkers.Stop` function not decrementing the worker group, for another day!

Reply:
I believe calling stop does decrement the worker group, but the wait group count is not exposed by go. see [this](https://github.com/golang/go/issues/7202)

Reply:
Oh awesome find, also very good to know that we can't check it in that way.

Thanks for looking into this!